src/OrchardCore.Modules/OrchardCore.Lists/Services/ContainerService.cs

```diff
@@ -283,6 +290,29 @@ public async Task<IEnumerable<ContentItem>> QueryContainedItemsAsync(string cont
                     return containedItems;
                 }
 
+                if (listPartFilter.DisplayText != null)
```

Comment:
Ennek a kódnak a `query.ListAsync()` előtt kellene lennie, mert akkor szűr SQL-ben. Most leránt mindent a DB-ből és memóriában szűr, ami nem jó.

Reply:
tehát az egész query.ListAsync() utáni rész a DisplayText-et ellenőrző if-en belülre kerüljön? Akkor a status-os switch kívül lesz?
Ha nem belülre kerül hanem csak simán fölé teszem a feltételem és az utána lévő szűrést akkor fel sem ismeri még a containedItems változót

Reply:
Nem ezt mondom, olvasd el kérlek még egyszer, amit írtam és vesd össze a kóddal.

Reply:
Az a gond hogy miután átnéztem pontosabban, úgy gondolom ezek a szűrőfeltételek több helyre is kellenének, mert úgy indul az egész feltétel a Task-ban, hogy megnézi, a pager.before, aztán pager.after null e? Ha csak egy oldalnyi item van akkor ugye ezekbe bele se fut, de leteszteltem úgy hogy több mint tíz blogpostot csináltam, és ha jól értem akkor 3 helyen ott kellene lennie displaytext-es és switch-es szűrésnek is. Mit gondolsz?
Ezt a szűrést nem lenne jobb egy function-ból meghívogatni valahogy?

Reply:
Ez pl. a `CreateOrderedContentIndexFilter()`-hez hasonlóan lehet egy metódusban.

Reply:
Ez a CreateOrderedContentIndexFilter mikor fut le? Mert ebben van a publishedonly és nem sikerül debuggolnom, hogy jó értéket adtam e paraméternek, mikor átírtam az enum miatt

Reply:
Hát akkor fut le, amikor az egyik hívási helyen meghívják :). Ha végigköveted a kódot, akkor látni fogod, ez mikor történhet. VS-ben továbbá a Ctrl+F12 megmondja, hol használnak egy metódust.